packages/core/src/DdSdkReactNativeConfiguration.tsx

```diff
@@ -62,6 +77,123 @@ export class DdSdkReactNativeConfiguration {
         readonly trackInteractions: boolean = false,
         readonly trackResources: boolean = false,
         readonly trackErrors: boolean = false,
-        readonly trackingConsent: TrackingConsent = TrackingConsent.GRANTED
+        readonly trackingConsent: TrackingConsent = DEFAULTS.trackingConsent
     ) {}
 }
+
+export type AutoInstrumentationConfiguration = {
+    readonly trackInteractions: boolean;
+    readonly trackResources: boolean;
+    readonly firstPartyHosts?: string[];
+    readonly resourceTracingSamplingRate?: number;
+    readonly trackErrors: boolean;
+};
+
+export type AutoInstrumentationParameters = {
+    readonly trackInteractions: boolean;
+    readonly trackResources: boolean;
+    readonly firstPartyHosts: string[];
+    readonly resourceTracingSamplingRate: number;
+    readonly trackErrors: boolean;
+};
```

Comment:
what is the difference between these? apart of 2 fields being nullable.

Reply:
One is the version that has default values as well. I think this can be improved.